### ISSUE TITLE
cabal install pdf-toolbox-document

### ISSUE THREAD

**Issue by TitusCornelius** [Yuras/pdf-toolbox#29]
Hey, I am trying to install pdf-toolbox on OSX/Mac and somehow the build is failing. Do you have any hints on how to solve this? Thanks in advance!

```
cabal install pdf-toolbox-document
Resolving dependencies...
Configuring pdf-toolbox-document-0.0.7.0...
Building pdf-toolbox-document-0.0.7.0...
Failed to install pdf-toolbox-document-0.0.7.0
Build log ( /Users/cornelius/.cabal/logs/pdf-toolbox-document-0.0.7.0.log ):
Configuring pdf-toolbox-document-0.0.7.0...
Building pdf-toolbox-document-0.0.7.0...
Preprocessing library pdf-toolbox-document-0.0.7.0...
[ 1 of 14] Compiling Prelude          ( compat/Prelude.hs, dist/build/Prelude.o )
[ 2 of 14] Compiling Pdf.Toolbox.Document.Internal.Util ( lib/Pdf/Toolbox/Document/Internal/Util.hs, dist/build/Pdf/Toolbox/Document/Internal/Util.o )
[ 3 of 14] Compiling Pdf.Toolbox.Document.Internal.Types ( lib/Pdf/Toolbox/Document/Internal/Types.hs, dist/build/Pdf/Toolbox/Document/Internal/Types.o )
[ 4 of 14] Compiling Pdf.Toolbox.Document.Encryption ( lib/Pdf/Toolbox/Document/Encryption.hs, dist/build/Pdf/Toolbox/Document/Encryption.o )
[ 5 of 14] Compiling Pdf.Toolbox.Document.Monad ( lib/Pdf/Toolbox/Document/Monad.hs, dist/build/Pdf/Toolbox/Document/Monad.o )
[ 6 of 14] Compiling Pdf.Toolbox.Document.Pdf ( lib/Pdf/Toolbox/Document/Pdf.hs, dist/build/Pdf/Toolbox/Document/Pdf.o )
[ 7 of 14] Compiling Pdf.Toolbox.Document.Document ( lib/Pdf/Toolbox/Document/Document.hs, dist/build/Pdf/Toolbox/Document/Document.o )
[ 8 of 14] Compiling Pdf.Toolbox.Document.Catalog ( lib/Pdf/Toolbox/Document/Catalog.hs, dist/build/Pdf/Toolbox/Document/Catalog.o )
[ 9 of 14] Compiling Pdf.Toolbox.Document.PageNode ( lib/Pdf/Toolbox/Document/PageNode.hs, dist/build/Pdf/Toolbox/Document/PageNode.o )
[10 of 14] Compiling Pdf.Toolbox.Document.Info ( lib/Pdf/Toolbox/Document/Info.hs, dist/build/Pdf/Toolbox/Document/Info.o )
[11 of 14] Compiling Pdf.Toolbox.Document.FontDict ( lib/Pdf/Toolbox/Document/FontDict.hs, dist/build/Pdf/Toolbox/Document/FontDict.o )

lib/Pdf/Toolbox/Document/FontDict.hs:51:36:
    Could not deduce (Functor m) arising from a use of ‘<$>’
    from the context (MonadPdf m, MonadIO m)
      bound by the type signature for
                 fontDictLoadInfo :: (MonadPdf m, MonadIO m) =>
                                     FontDict -> PdfE m FontInfo
      at lib/Pdf/Toolbox/Document/FontDict.hs:47:21-74
    Possible fix:
      add (Functor m) to the context of
        the type signature for
          fontDictLoadInfo :: (MonadPdf m, MonadIO m) =>
                              FontDict -> PdfE m FontInfo
    In the expression:
      FontInfoComposite <$> loadFontInfoComposite fontDict
    In a case alternative:
        FontType0 -> FontInfoComposite <$> loadFontInfoComposite fontDict
    In a stmt of a 'do' block:
      case subtype of {
        FontType0 -> FontInfoComposite <$> loadFontInfoComposite fontDict
        FontType3
          -> do { fi <- loadFontInfoSimple fontDict;
                  Array arr <- lookupDict "FontMatrix" fontDict >>= deref
                               >>= fromObject;
                  .... }
        _ -> FontInfoSimple <$> loadFontInfoSimple fontDict }

lib/Pdf/Toolbox/Document/FontDict.hs:154:30:
    Could not deduce (Functor m) arising from a use of ‘<$>’
    from the context (MonadPdf m)
      bound by the type signature for
                 loadEncodingDifferences :: MonadPdf m =>
                                            Dict -> PdfE m [(Word8, ByteString)]
      at lib/Pdf/Toolbox/Document/FontDict.hs:145:28-77
    Possible fix:
      add (Functor m) to the context of
        the type signature for
          loadEncodingDifferences :: MonadPdf m =>
                                     Dict -> PdfE m [(Word8, ByteString)]
    In a stmt of a 'do' block: n' <- fromIntegral <$> intValue n
    In the expression:
      do { n' <- fromIntegral <$> intValue n;
           go [] n' rest }
    In a case alternative:
        (ONumber n : rest)
          -> do { n' <- fromIntegral <$> intValue n;
                  go [] n' rest }

lib/Pdf/Toolbox/Document/FontDict.hs:177:30:
    Could not deduce (Functor m) arising from a use of ‘<$>’
    from the context (MonadPdf m, MonadIO m)
      bound by the type signature for
                 loadUnicodeCMap :: (MonadPdf m, MonadIO m) =>
                                    Dict -> PdfE m (Maybe UnicodeCMap)
      at lib/Pdf/Toolbox/Document/FontDict.hs:167:20-80
    Possible fix:
      add (Functor m) to the context of
        the type signature for
          loadUnicodeCMap :: (MonadPdf m, MonadIO m) =>
                             Dict -> PdfE m (Maybe UnicodeCMap)
    In a stmt of a 'do' block:
      content <- mconcat <$> liftIO (Streams.toList is)
    In the expression:
      do { Stream _ is <- streamContent ref s;
           content <- mconcat <$> liftIO (Streams.toList is);
           case parseUnicodeCMap content of {
             Left e -> throwE $ UnexpectedError $ "can't parse cmap: " ++ show e
             Right cmap -> return $ Just cmap } }
    In a case alternative:
        OStream s
          -> do { Stream _ is <- streamContent ref s;
                  content <- mconcat <$> liftIO (Streams.toList is);
                  case parseUnicodeCMap content of {
                    Left e -> throwE $ UnexpectedError $ "can't parse cmap: " ++ show e
                    Right cmap -> return $ Just cmap } }
cabal: Error: some packages failed to install:
pdf-toolbox-document-0.0.7.0 failed during the building phase. The exception
was:
ExitFailure 1
```

**Comment by Yuras** [comment]
Fixed via b44b03c91a987ffbad8a2af3a82b99109c6231f4

Thank you for reporting the issue!

**Comment by TitusCornelius** [comment]
thank you for fixing it so quickly!
